src/Arcus.WebApi.Security.Authentication/SharedAccessKey/SharedAccessKeyAuthenticationFilter.cs

```diff
@@ -53,27 +55,20 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
             Guard.For<ArgumentException>(() => context.HttpContext.Request.Headers == null, "Invalid action context given without any HTTP request headers");
             Guard.For<ArgumentException>(() => context.HttpContext.RequestServices == null, "Invalid action context given without any HTTP request services");
 
+            string foundSecret = await GetAuthorizationSecret(context);
+
             if (context.HttpContext.Request.Headers
                        .TryGetValue(_headerName, out StringValues requestSecretHeaders))
             {
-                ISecretProvider userDefinedSecretProvider = 
-                    context.HttpContext.RequestServices.GetService<ICachedSecretProvider>()
-                    ?? context.HttpContext.RequestServices.GetService<ISecretProvider>();
-                
-                if (userDefinedSecretProvider == null)
-                {
-                    throw new KeyNotFoundException(
-                        $"No configured {nameof(ICachedSecretProvider)} or {nameof(ISecretProvider)} implementation found in the request service container. "
-                        + "Please configure such an implementation (ex. in the Startup) of your application");
-                }
 
-                string foundSecret = await userDefinedSecretProvider.Get(_secretName);
-                if (foundSecret == null)
+                if (requestSecretHeaders.Any(headerValue => !String.Equals(headerValue, foundSecret)))
                 {
-                    throw new SecretNotFoundException(_secretName);
+                    context.Result = new UnauthorizedResult();
                 }
-
-                if (requestSecretHeaders.Any(headerValue => !String.Equals(headerValue, foundSecret)))
+            }
+            else if (context.HttpContext.Request.Query.ContainsKey(_queryParameterName))
```

Comment:
I would suggest not to use an `else if`, if the API consumer specifies both the header *and* query parameter, they both have to match.

Reply:
I thought the idea was to have one OR the other, not both together which also seems a bit odd to me. Could this actually be a requirement?

Reply:
The API can configure to support both header & query parameter authentication, but it's still up to the client to choose which approach they want to use.

If you ask me, the following rules apply:
- API consumer specifies header without query parameter, header must match
- API consumer specifies query parameter without header, query parameter must match
- API consumer specifies query parameter & header, query parameter & header must match
- API consumer specifies nothing, instant revoke

_(All scenarios should also be covered with tests)_

Reply:
@tomkerkhove This has been updated in latest PR

Reply:
@tomkerkhove has this been resolved now?